react/react1/week3/reactweek3submit/reactsubmit/src/ViewShift.js

```diff
@@ -0,0 +1,57 @@
+import "./App.css";
+import React from "react";
+import FancyBorder from "./FancyBorder";
+
+export function ViewShift({
+  id,
+  name,
+  startTime,
+  endTime,
+  deleteShift,
+  searchName,
+  searchStart,
+  searchEnd,
+}) {
+  const endHoursToMinutes = endTime.split(":");
+  const totalEndMinutes =
+    Number(endHoursToMinutes[0]) * 60 + Number(endHoursToMinutes[1]);
+  const startHoursToMinutes = startTime.split(":");
+  const totalStartMinutes =
+    Number(startHoursToMinutes[0]) * 60 + Number(startHoursToMinutes[1]);
+  const workingMinutes = totalEndMinutes - totalStartMinutes;
+  const hours = Math.floor(workingMinutes / 60);
+  const minutes = workingMinutes % 60;
+  const total = hours + ":" + minutes;
+  console.log(total);
```

Comment:
This calculation works. Nice!

It could be a bit easier to read though. Try to use meaningful and precise variable names when possible.

```suggestion
  const startTimeArray = startTime.split(":");
  const startTimeInMinutes = Number(startTimeArray[0]) * 60 + Number(startTimeArray[1]);
  const endTimeArray = endTime.split(":");
  const endTimeInMinutes = Number(endTimeArray[0]) * 60 + Number(endTimeArray[1]);
  const totalMinutes = endTimeInMinutes - startTimeInMinutes;
  const hours = Math.floor(totalMinutes / 60);
  const minutes = totalMinutes % 60;
  const timeWorked = hours + ":" + minutes;
```

Reply:
done